compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -3522,6 +3548,22 @@ private ParserRuleContext getNextRuleForExpr() {
         return ParserRuleContext.EXPRESSION_RHS;
     }
 
+    private ParserRuleContext getNextRuleForExprStartsWithIdentifier() {
+        ParserRuleContext parentCtx;
+        parentCtx = getParentContext();
+        if (parentCtx == ParserRuleContext.CONSTANT_EXPRESSION) {
+            endContext();
+            return getNextRuleForConstExpr();
+        }
+        return getNextRuleForExprWithFuncCallEnabled();
+    }
+
+    private ParserRuleContext getNextRuleForExprWithFuncCallEnabled() {
+        // Here we start a new context so that expr-rhs will allow function calls
+        startContext(ParserRuleContext.FUNCTION_CALL_START);
```

Comment:
Can we do it without starting a context? because it is also an expensive operation. Should use that only if theres no other way.

What if we give an alias, say "VARIABLE_REF_RHS", and that will call the `seekMatchInExpressionRhs` with the flag on/off